### PR TITLE
chore: add missing unix dependencies

### PR DESCRIPTION
**Pull request by emillon**
This suppresses warnings with ocaml >= 5.

### CODE REVIEW

**Comment by emillon** [comment]
2 notes:
- there's one to fix in our vendored fork of opam
- I don't understand why the one in `dune_findlib` is necessary, since `Unix` is not referenced in that library

**Comment by rgrinberg** [comment]
@nojb or @dra27 why does the warning fire in this case?

In any case, I would rather disable the warning where it's not applicable rather than adding a dependency on unix.

**Comment by emillon** [comment]
for that particular one we can investigate - the other ones are legitimate (the libraries use Unix).
Something that's possible is that it's triggering at a distance from another library that uses unix in an undeclared way but this is only reported when linking a dependent library.

File: src/dune_findlib/dune [hunk 1]
```diff
@@ -1,3 +1,3 @@
 (library
  (name dune_findlib)
- (libraries stdune dune_lang dune_engine memo ocaml dune_meta_parser))
+ (libraries stdune dune_lang dune_engine memo ocaml dune_meta_parser unix))
```
Comment (rgrinberg) [inline src/dune_findlib/dune:3]: As discussed, we should disable the warning for this library because it doesn't use Unix.

Comment (emillon) [reply src/dune_findlib/dune]: done

**Comment by nojb** [comment]
> Something that's possible is that it's triggering at a distance from another library that uses unix in an undeclared way but this is only reported when linking a dependent library.

I think this is the explanation: one of the dependencies must use Unix without declaring it.

**Comment by emillon** [comment]
I got to the bottom of this. In this line:

https://github.com/ocaml/dune/blob/faa389a09b20d81f6f339ed1ef20d3fd4349c9bd/src/dune_findlib/config.ml#L46

the `_` has type `Dune_engine.Import.Unix_error.Detailed.t`, which expands to `Unix.error * string * string`. So while the source code does not refer to the `Unix` module, it's still "depending" on that module.

**Comment by Alizter** [comment]
@emillon Does that still happen if you make it `Result.get_ok`?

**Comment by emillon** [comment]
using `Result.to_option` does not work. I don't think that `_` is the problem, but rather that `Unix` appears in the type of a function we use.

**Review by rgrinberg — APPROVED** [review]
LGTM. I would suggest  to add a type annotation to this `_` so that it's obvious

**Comment by emillon** [comment]
I added an annotation. re exporting unix from dune_engine would also work.

**Pull request merged**